File: test-utility/src/main/java/com/mobilesolutionworks/android/controller/test/util/ResumeLatch.java
//package com.mobilesolutionworks.android.controller.test.util;
//
//import android.app.Activity;
//import android.app.Application;
//import android.os.Bundle;
//
//import java.util.concurrent.CountDownLatch;
//import java.util.concurrent.TimeUnit;
//
///**
// * Created by yunarta on 12/3/17.
// */
//public class ResumeLatch implements Application.ActivityLifecycleCallbacks {
//
//    private CountDownLatch mLatch;
//
//    public ResumeLatch() {
//        mLatch = new CountDownLatch(1);
//    }
//
//    public void await() throws InterruptedException {
//        mLatch.await(5, TimeUnit.SECONDS);
//        reset();
//    }
//
//    public void reset() {
//        mLatch = new CountDownLatch(1);
//    }
//
//    @Override
//    public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
//
//    }
//
//    @Override
//    public void onActivityStarted(Activity activity) {
//
//    }
//
//    @Override
//    public void onActivityResumed(Activity activity) {
//        mLatch.countDown();
//    }
//
//    @Override
//    public void onActivityPaused(Activity activity) {
//
//    }
//
//    @Override
//    public void onActivityStopped(Activity activity) {
//
//    }
//
//    @Override
//    public void onActivitySaveInstanceState(Activity activity, Bundle outState) {
//
//    }
//
//    @Override
//    public void onActivityDestroyed(Activity activity) {
//
//    }
//}

File: test-utility/src/main/java/com/mobilesolutionworks/android/controller/test/util/ResumeLatch.kt
package com.mobilesolutionworks.android.controller.test.util

import android.app.Activity
import android.app.Application
import android.os.Bundle

import java.util.concurrent.CountDownLatch
import java.util.concurrent.TimeUnit

/**
 * Created by yunarta on 12/3/17.
 */
class ResumeLatch : Application.ActivityLifecycleCallbacks {

    private var mLatch: CountDownLatch? = null

    init {
        mLatch = CountDownLatch(1)
    }

    @Throws(InterruptedException::class)
    fun await() {
        mLatch!!.await(5, TimeUnit.SECONDS)
        reset()
    }

    fun reset() {
        mLatch = CountDownLatch(1)
    }

    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle) {

    }

    override fun onActivityStarted(activity: Activity) {

    }

    override fun onActivityResumed(activity: Activity) {
        mLatch!!.countDown()
    }

    override fun onActivityPaused(activity: Activity) {

    }

    override fun onActivityStopped(activity: Activity) {

    }

    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {

    }

    override fun onActivityDestroyed(activity: Activity) {

    }
}
